nanoFramework.System.Net.Http/Http/System.Net.HttpWebRequest.cs

```diff
@@ -1433,9 +1457,16 @@ private InputNetworkStreamWrapper EstablishConnection(Uri proxyServer, Uri targe
                 {
                     IPEndPoint remoteEP = new IPEndPoint(address, proxyServer.Port);
                     socket.Connect((EndPoint)remoteEP);
-                }
-                catch (SocketException e)
+                    // skigrinder - check the newly created socket
```

Comment:
If the socket was just connected, why do we need to poll it right away to check the connection?

Reply:
Call to Poll() removed.
Socket is now closed if exception is caught.
This fixes a memory leak issue.